# Copyright 2026 Qore Technologies, s.r.o.; MIT.
# Match the structured error independently of Error.toString formatting.
--- a/test/parallel/test-process-euid-egid.js
+++ b/test/parallel/test-process-euid-egid.js
@@ -1,3 +1,4 @@
+// Copyright 2026 Qore Technologies, s.r.o.; SPDX-License-Identifier: MIT
 'use strict';
 
 const common = require('../common');
@@ -48,7 +49,10 @@
 
   assert.throws(() => {
     process.seteuid('nobody');
-  }, /^Error: (?:EPERM, .+|User identifier does not exist: nobody)$/);
+  }, {
+    code: /^(?:EPERM|ERR_UNKNOWN_CREDENTIAL)$/,
+    message: /^(?:EPERM, .+|User identifier does not exist: nobody)$/
+  });
 
   return;
 }
